### ISSUE TITLE
allow hr separated sections to share titles/subtitles

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [DavidAnson/markdownlint#438]
when writing markdown presentations with markdown using hr as separator its  sometimes helpful to repeat a larger title
followup to #437 

```markdown
## topic
### part 1
---

## topic
### part 2
-----

## topic b
### part 1
---
## topic b
### part 2
```

i have that pattern in presentatiosn that have short code snippets followed by output snipped

**Comment by RonnyPfannschmidt** [comment]
It's what i currently do,

I do prefer to minimise the linter ignores per project

**Comment by NudeNULL** [comment]
I am voting it up as well, why showing a warning if it specifically adheres to the rules for changelogs
![image](https://github.com/DavidAnson/markdownlint/assets/15786742/6cf6f4bd-035b-47fc-88a3-50e075508269)

**Comment by DavidAnson** [comment]
@NudeNULL, the changelog scenario is already supported via the `siblings_only` parameter: https://github.com/DavidAnson/markdownlint/blob/main/doc/md024.md

**Comment by NudeNULL** [comment]
Oh cool thanks! Thats very handy.
